.evergreen/linker-tests.sh

```diff
@@ -58,7 +58,7 @@ mkdir cmake-build
 cd cmake-build
 INSTALL_PATH="$(system_path $linker_tests_root/install/bson1)"
 SRC_PATH="$(system_path ../)"
-$CMAKE -DBUILD_VERSION=1.18.0-pre -DENABLE_MONGOC=OFF -DCMAKE_BUILD_TYPE=RelWithDebInfo $ADDITIONAL_CMAKE_FLAGS -DCMAKE_INSTALL_PREFIX="$INSTALL_PATH" "$SRC_PATH"
```

Comment:
It is not necessary to specify `BUILD_VERSION` since this is included in the `VERSION_CURRENT` file.